File: kardex-service/src/main/java/com/example/kardex_service/services/KardexService.java
package com.example.kardex_service.services;

import com.example.kardex_service.entities.KardexEntity;
import com.example.kardex_service.repository.KardexRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.ArrayList;

@Service
public class KardexService {
    @Autowired
    KardexRepository kardexRepository;

    // Find Kardex's
    public ArrayList<KardexEntity> getKardexes(){
        return (ArrayList<KardexEntity>) kardexRepository.findAll();
    }

    // Create Kardex
    public KardexEntity createKardex(KardexEntity kardexEntity){
        return kardexRepository.save(kardexEntity);
    }

    // Find Kardex by Id
    public KardexEntity getKardexById(Long id){
        return kardexRepository.findById(id).get();
    }

    // Update Kardex
    public KardexEntity updateKardex(KardexEntity kardexEntity){
        return kardexRepository.save(kardexEntity);
    }

    // Delete Kardex
    public boolean deleteKardex(Long id) throws Exception{
        try {
            kardexRepository.deleteById(id);
            return true;
        }catch (Exception e){
            throw new Exception(e.getMessage());
        }
    }
}

File: frontend/src/services/kardex.service.js
import httpKardex from "../http-common";

const getAllKardex = () => {
    return httpKardex.get('/kardex-service/kardex/');
}           

const createKardex = data => {
    return httpKardex.post('/kardex-service/kardex/', data);
}

const getKardexById = id => {
    return httpKardex.get(`/kardex-service/kardex/${id}`);
}

const updateKardex = (data) => {
    return httpKardex.put('/kardex-service/kardex/', data);
}

const deleteKardex = id => {
    return httpKardex.delete(`/kardex-service/kardex/${id}`);
}

const getKardexByDate = (startDate, endDate) => {
    return httpKardex.get(`/kardex-service/kardex/byDate/${startDate}/${endDate}`);
}

export default { getAllKardex, createKardex, getKardexById, updateKardex, deleteKardex, getKardexByDate };